libs/IMUKit/tests/IMUKit_test.cpp

```diff
@@ -0,0 +1,163 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include "IMUKit.h"
+
+#include "CoreAccelerometer.h"
+#include "CoreGyroscope.h"
+#include "ThisThread.h"
+#include "gtest/gtest.h"
+#include "mocks/leka/LSM6DSOX.h"
+#include "stubs/leka/EventLoopKit.h"
+
+using namespace leka;
+
+using testing::ReturnRef;
+
+class IMUKitTest : public ::testing::Test
+{
+  protected:
+	IMUKitTest() = default;
+
+	void SetUp() override { imukit.init(); }
+	// void TearDown() override {}
+
+	stub::EventLoopKit stub_event_loop {};
+
+	mock::LSM6DSOX mock_lsm6dsox;
+
+	CoreAccelerometer accel {mock_lsm6dsox};
+	CoreGyroscope gyro {mock_lsm6dsox};
+
+	IMUKit imukit {stub_event_loop, accel, gyro};
+
+	interface::LSM6DSOX::SensorData data
+	{
+		{0.F, 0.F, 0.F}, { 0.F, 0.F, 0.F }
+	};
+};
+
+TEST_F(IMUKitTest, initialization)
+{
+	ASSERT_NE(&imukit, nullptr);
+}
+
+TEST_F(IMUKitTest, computeAnglesNullAccelerations)
+{
+	data = {{0.F, 0.F, 0.F}};
+
+	EXPECT_CALL(mock_lsm6dsox, getData).WillRepeatedly(ReturnRef(data));
+
+	for (auto i = 0; i < 100; ++i) {
+		imukit.computeAngles();
+	}
+
+	auto [pitch, roll, yaw] = imukit.getAngles();
+
+	EXPECT_EQ(pitch, 0);
+	EXPECT_EQ(roll, 0);
+	EXPECT_EQ(yaw, 180);
+}
+
+TEST_F(IMUKitTest, computeAngles1GAccelerationOnXAxis)
+{
+	data = {{1000.F, 0.F, 0.F}};
+
+	EXPECT_CALL(mock_lsm6dsox, getData).WillRepeatedly(ReturnRef(data));
+
+	for (auto i = 0; i < 100; ++i) {
+		imukit.computeAngles();
+	}
+
+	auto [pitch, roll, yaw] = imukit.getAngles();
+
+	EXPECT_TRUE(pitch < -80 && pitch > -100);
```

Comment:
Details but can be set this way to be more comfortable to read

```suggestion
	EXPECT_TRUE(-100 < pitch && pitch < -80);
```